File: src/main/scala/syntax/scala/TypeAliases.scala
package syntax.scala

class TypeAliases {
  
  type Versao = Int
  type Significado  = String
  type Significados = Map[Versao,Significado]
  type Palavra = String
  type Dicionario = Map[Palavra,Significados]
  
  val dics:Dicionario = Map()
  def lookup(palavra:Palavra,versao:Versao) = 
    dics.get(palavra)
        .flatMap{_.get(versao)}
        .getOrElse(throw new IllegalArgumentException)
}

File: src/main/java/syntax/java/TypeAliases.java
package syntax.java;

import java.util.HashMap;
import java.util.Map;

public class TypeAliases {
	
	Map<String,Map<Integer,String>> dics = new HashMap<>();
	
	public String lookup(String palavra, Integer versao) {
		if(dics.containsKey(palavra)){
			Map<Integer,String> dict = dics.get(palavra);
			if (dict.containsKey(versao)){
				return dict.get(versao);
			}
		}
		throw new IllegalArgumentException();
	}
	
	
}
